### PR TITLE
Add Swerve Subsystem and Config Files

### PR DESCRIPTION
**Pull request by AadityaAgni**
Created SwerveSubsystem.java, which is the subsystem for the swerve drive base. While we heavily borrowed from last year's code and other team's ideas, here are the main differences:

1. SwerveSubsytem uses SwerveConfig.java, which contains the initialization for SwerveSubsystem, the ID's, and kinematics
2. Modules are now contained in SwerveConfig, but in Swerve Subsystem it is a list of modules

### CODE REVIEW

**Comment by tcdoan** [comment]
Going forward let us add @FRC-1294/software as reviewer for every Pull Requests

**Review by ConfettiRaindrop — APPROVED** [review]

**Comment by tcdoan** [comment]
@AadityaAgni @bredisrising Delete Constants.java

File: src/main/java/frc/robot/constants/SwerveConstants.java [hunk 1]
```diff
@@ -4,23 +4,34 @@
...
+  public static final double kPhysicalMaxSpeedMPS = 4.0;
 
+  public static final double kTeleMaxSpeedMPS = 4.0;
+  public static final double kAutoMaxSpeedMPS = 3.0;
+
+  // Encoder Offsets
+  public static double frontRightRotEncoderOffset = 0;
+  public static double frontLeftRotEncoderOffset = 0;
+  public static double backRightRotEncoderOffset = 0;
+  public static double backLeftRotEncoderOffset = 0;
```
Comment (tcdoan) [inline src/main/java/frc/robot/constants/SwerveConstants.java:34]: @AadityaAgni  prefix "k" to all constants

File: src/main/java/frc/robot/subsystems/SwerveModule.java [hunk 2]
```diff
@@ -33,14 +30,12 @@ public class SwerveModule {
     private AnalogEncoder _rotEncoder;
     private RelativeEncoder _transEncoder;
 
-    public SwerveModule(int rotID, int transID, int rotEncoderID, int transEncoderID,
-            double rotEncoderOffset, boolean rotInverse, boolean transInverse, PIDController rotPID,
-            PIDController transPID) {
+    public SwerveModule(int rotID, int transID, int rotEncoderID, double rotEncoderOffset,
```
Comment (tcdoan) [inline src/main/java/frc/robot/subsystems/SwerveModule.java:33]: Too many parameters. Abstract them into SwerveModuleConfig object that can be loaded from a json file

**Review by tcdoan — CHANGES_REQUESTED** [review]

**Review by tcdoan — APPROVED** [review]
Please address the SwerveModule constructor having many parameters in the next PR

**Review by Powerlax — APPROVED** [review]
LGMT!

**Pull request merged**